src/main/kotlin/org/hyacinthbots/lilybot/extensions/events/ModThreadInviting.kt

```diff
@@ -0,0 +1,56 @@
+package org.hyacinthbots.lilybot.extensions.events
+
+import com.kotlindiscord.kord.extensions.checks.anyGuild
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.event
+import dev.kord.core.behavior.channel.createMessage
+import dev.kord.core.behavior.edit
+import dev.kord.core.event.channel.thread.ThreadChannelCreateEvent
+import dev.kord.core.supplier.EntitySupplyStrategy
+import kotlinx.coroutines.delay
+import org.hyacinthbots.lilybot.database.collections.AutoThreadingCollection
+import org.hyacinthbots.lilybot.database.collections.ModerationConfigCollection
+
+class ModThreadInviting : Extension() {
+	override val name: String = "mod-thread-inviting"
+
+	override suspend fun setup() {
+		event<ThreadChannelCreateEvent> {
+			check {
+				anyGuild()
+				failIf {
+					event.channel.ownerId == kord.selfId ||
+							event.channel.member != null
+				}
+			}
+
+			action {
+				// fixme this event fires twice for some unknown reason so this is a workaround
+				delay(1000)
+				if (event.channel.getLastMessage()?.withStrategy(EntitySupplyStrategy.rest) != null) return@action
+
+				val channel = event.channel
+
+				if (AutoThreadingCollection().getSingleAutoThread(channel.id) != null) return@action
+
+				val config = ModerationConfigCollection().getConfig(channel.guildId) ?: return@action
+
+				if (!config.enabled || config.role == null) return@action
+
+				val moderatorRole = channel.guild.getRoleOrNull(config.role) ?: return@action
+
+				if (!moderatorRole.mentionable) return@action
+
+				val message = channel.createMessage {
+					content = "Placeholder message"
+				}
+
+				message.edit {
+					content = moderatorRole.mention
+				}
+
+				message.delete()
```

Comment:
Why must the mention be deleted?

Reply:
Leaving it around adds clutter and tends to confuse people. We can run a poll about it if you'd like, but I think it's better to just have the system work without most users knowing.

Reply:
I think a poll is suitable. I disagree with deleting tbh

Reply:
The new system won the poll unanimously, so we're deleting the message.